operation/fhir-operation-validate/src/main/java/com/ibm/fhir/operation/validate/ValidateOperation.java

```diff
@@ -46,14 +47,26 @@ protected Parameters doInvoke(FHIROperationContext operationContext, Class<? ext
             String versionId, Parameters parameters, FHIRResourceHelpers resourceHelper, SearchHelper searchHelper) throws FHIROperationException {
         try {
             Parameter resourceParameter = getParameter(parameters, "resource");
-            if (resourceParameter == null) {
+            String resourceTypeName = resourceType.getSimpleName();
+            Resource resource;
+            // @implNote if $validate is on resource-instance level, fetch the resource from database.
+            if (FHIROperationContext.Type.INSTANCE.equals(operationContext.getType()) && resourceParameter == null) {
+                resource = resourceHelper.doRead(resourceTypeName, logicalId).getResource();
+                if (resource == null) {
+                    throw buildExceptionWithIssue(resourceTypeName + " with id '" + logicalId + "' was not found", IssueType.NOT_FOUND);
+                }
+            } else if (resourceParameter == null) {
                 throw buildExceptionWithIssue("Input parameter 'resource' is required for the $validate operation", IssueType.INVALID);
+            } else {
+                resource = resourceParameter.getResource();
             }
-            Resource resource = resourceParameter.getResource() ;
-
+            
             List<OperationOutcome.Issue> issues;
             Parameter profileParameter = getParameter(parameters, "profile");
             Parameter modeParameter = getParameter(parameters, "mode");
+            
+            // @implNote validation for only valid "mode" codes are accepted by the $validate operation
```

Comment:
same comment

Reply:
```suggestion
            // Ensure that the passed 'mode' parameter is valid.
```

Reply:
this has been fixed. removed "@implNote"